letter-shop.md

```diff
@@ -0,0 +1,232 @@
+# Letter Shop Tutorial
+
+## What you need before you start:
+
+* a laptop with an internet connection and NodeJS installed
+* basic knowledge of the command line terminal
+* basic knowledge of JavaScript, including Promises and Buffers
+* (optional) the concept of one-way hash functions
+
+## What you'll learn:
+
+* the concept Hash Time Lock Agreements (HTLAs)
+* the Ledger Plugin Interface (LPI)
+* how to build a paid web app using Interledger!
+
+## Step 1:
+
+Save the following JavaScript as `shop.js`:
+
+```js
+const http = require('http')
+const crypto = require('crypto')
+const Plugin = require('ilp-plugin-xrp-escrow')
+function base64 (buf) { return buf.toString('base64').replace(/\+/g, '-').replace(/\//g, '_').replace(/=+$/, '') }
+
+let fulfillments = {}
+let letters = {}
+
+const plugin = new Plugin({
+  secret: 'ssGjGT4sz4rp2xahcDj87P71rTYXo',
+  account: 'rrhnXcox5bEmZfJCHzPxajUtwdt772zrCW',
+  server: 'wss://s.altnet.rippletest.net:51233',
+  prefix: 'test.crypto.xrp.'
+})
+
+plugin.connect().then(function () {
+  plugin.on('incoming_prepare', function (transfer) {
+    plugin.fulfillCondition(transfer.id, fulfillments[transfer.executionCondition]).catch(function () {})
+  })
+
+  http.createServer(function (req, res) {
+    if (letters[req.url.substring(1)]) {
+      res.end('Your letter: ' + letters[req.url.substring(1)])
+    } else {
+      const secret = crypto.randomBytes(32)
+      const fulfillment = base64(secret)
+      const condition = base64(crypto.createHash('sha256').update(secret).digest())
```

Comment:
nit: put some of the function calls on different lines or make the code area wider so the whole line fits without scrolling

Reply:
yeah, the code area is annoyingly narrow, but then also having to use linebreaks everywhere hurts the readability. I've gone back and forth on that. Maybe I can make the code area wider somehow.

Reply:
Changed it to Modernist now, which gives 23 more chars width (up from 61 to 84). We could get another 8 chars by switching to Cayman, and that theme may fit better in the Interledger house style, but probably looks less fancy for tutorials.